modules/idlgen/plugin/src/main/scala/IdlGenPlugin.scala

```diff
@@ -28,33 +28,70 @@ object IdlGenPlugin extends AutoPlugin {
     lazy val idlGen: TaskKey[Unit] =
       taskKey[Unit]("Generates IDL files from freestyle-rpc service definitions")
 
-    lazy val sourceDir: SettingKey[File] =
+    lazy val srcGen: TaskKey[Unit] =
```

Comment:
We might want to make this task as `TaskKey[Seq[File]]`, in order to be able to add the following configuration:

`sourceGenerators in Compile += (srcGen in Compile).taskValue`

triggering the `srcGen` task into the compile phase.